File: src/main/java/com/skocur/imagecipher/encrypters/SingleColorEncryption.java
package com.skocur.imagecipher.encrypters;

import com.skocur.imagecipher.Config;
import com.skocur.imagecipher.plugin.lib.annotations.IcAlgorithmSpecification;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import org.jetbrains.annotations.NotNull;

import java.awt.*;
import java.io.IOException;
import java.util.ArrayList;

@IcAlgorithmSpecification(algorithmName = "Single Color Encryption")
public class SingleColorEncryption extends Encrypter {

  private ArrayList<Integer> asciiChars;
  private static final Logger logger = LogManager.getLogger();

  public SingleColorEncryption(@NotNull String fileName) throws IOException {
    super(fileName);
    asciiChars = new ArrayList<>();
  }

  @Override
  public void encrypt(@NotNull String text) {
    logger.debug("Encrypting: " + text);
    for (char c : text.toCharArray()) {
      asciiChars.add((int) c);
    }

    for (int i = 0, index = 0; i < getImageWidth(); i++) {
      if (i % Config.SPACING_CIPHER == 0 && index < asciiChars.size()) {
        Color greenShade = new Color(1, asciiChars.get(index), 1);
        image.setRGB(i, Config.IMAGE_MARGIN_TOP, greenShade.getRGB());
        index++;
      }
    }
  }
}

File: app/src/main/kotlin/com/imagecipher/app/encrypters/SingleColorEncryption.kt
package com.imagecipher.app.encrypters

import com.imagecipher.app.Config
import com.imagecipher.app.plugin.lib.annotations.IcAlgorithmSpecification
import org.apache.logging.log4j.LogManager
import java.awt.Color

@IcAlgorithmSpecification(algorithmName = "Single Color Encryption")
class SingleColorEncryption(fileName: String) : Encrypter(fileName) {

    private val asciiChars: MutableList<Int> = mutableListOf()

    companion object {
        private val logger = LogManager.getLogger()
    }

    override fun encrypt(text: String) {
        logger.debug("Encrypting: $text")
        for (c in text) {
            asciiChars.add(c.code)
        }

        var index = 0
        for (i in 0 until getImageWidth()) {
            if (i % Config.SPACING_CIPHER == 0 && index < asciiChars.size) {
                val greenShade = Color(1, asciiChars[index], 1)
                image.setRGB(i, Config.IMAGE_MARGIN_TOP, greenShade.rgb)
                index++
            }
        }
    }
}
